### PR TITLE
examples: add binary tree

### PR DESCRIPTION
**Pull request by james-d-mitchell**
~~Missing tests and doc~~

### CODE REVIEW

File: doc/examples.xml [hunk 1]
```diff
@@ -711,12 +711,15 @@ true
   <Oper Name="BookGraph" Arg="[filt, ]m"/>
   <Returns>A digraph.</Returns>
   <Description>
-    The <E>Book digraph</E> is the Cartesian product of a complete digraph with 2 vertices
-    (as the "Book Spine") and the <M><A>m</A>+1</M> star digraph (as the "Pages").
-    For more details on the Book digraph please refer to <URL>https://mathworld.wolfram.com/BookGraph.html</URL>.<P/>
-
-    See <Ref Func="DigraphCartesianProduct" Label="for a positive number of digraphs"/>,
-    <Ref Oper="CompleteDigraph"/>, and <Ref Oper="StarGraph"/>. <P/>
+    The <E>Book digraph</E> is the Cartesian product of a complete digraph with
```
Comment (wilfwilson) [inline doc/examples.xml:714]: Reminder to self: graph/digraph naming mismatch

File: doc/main.xml [hunk 2]
```diff
@@ -24,6 +24,7 @@
      <!ENTITY datastructures "<Alt Not='Text'><URL
        Text='datastructures'>https://github.com/gap-packages/datastructures</URL></Alt><Alt
    Only='Text'><Package>datastructures</Package></Alt>">
+     <!ENTITY STANDARD_FILT_TEXT "If the optional first argument <A>filt</A> is present, then this should specify the category or representation the digraph being created will belong to. For example, if <A>filt</A> is <Ref Filt='IsMutableDigraph'/>, then the digraph being created will be mutable, if <A>filt</A> is <Ref Filt='IsImmutableDigraph'/>, then the digraph will be immutable.  If the optional first argument <A>filt</A> is not present, then <Ref Filt='IsImmutableDigraph'/> is used by default.<P/>">
```
Comment (wilfwilson) [inline doc/main.xml:27]: Yay! Thanks for adding this.

File: doc/examples.xml [hunk 3]
```diff
@@ -743,3 +746,26 @@ true
...
+
+<#GAPDoc Label="BinaryTree">
+<ManSection>
+  <Oper Name="BinaryTree" Arg="[filt, ]m"/>
+  <Returns>A digraph.</Returns>
+  <Description>
+    This function returns a binary tree of depth <A>m</A> with <C>2 ^ <A>m</A>
+    - 1</C> vertices. All edges are directed towards the root of the tree.
+    <P/>
+
```
Comment (wilfwilson) [inline doc/examples.xml:758]: Do you think this is worth adding? It makes it a bit more explicit about which vertices are which in the tree.
```suggestion
    Note that <C>BinaryTree(<A>m</A>)</C> is the induced subdigraph of
    <C>BinaryTree(<A>m</A>+1)</C> on the vertices <C>[1..2^(<A>m</A>-1)]</C>.
    <P/>
    
```

File: doc/examples.xml [hunk 4]
```diff
@@ -743,3 +746,26 @@ true
...
 </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="BinaryTree">
+<ManSection>
+  <Oper Name="BinaryTree" Arg="[filt, ]m"/>
+  <Returns>A digraph.</Returns>
+  <Description>
+    This function returns a binary tree of depth <A>m</A> with <C>2 ^ <A>m</A>
+    - 1</C> vertices. All edges are directed towards the root of the tree.
```
Comment (wilfwilson) [inline doc/examples.xml:756]:
```suggestion
    - 1</C> vertices. All edges are directed towards the root of the tree,
    which is vertex <C>1</C>.
```

File: doc/examples.xml (hunk 1 above)
Comment (wilfwilson) [reply doc/examples.xml:714]: I'll make a separate note to myself to check the manual for things like this and fix them. (Both the operation being 'BookGraph' but the documentation saying 'Book digraph', and also the use of capitalisation in this entry.

**Review by wilfwilson — COMMENTED** [review]
Nice, thanks for this. I've got a couple of suggestions of additions for the documentation, that's all.

**Review by wilfwilson — APPROVED** [review]
Sorry, meant to approve, feel free to adopt my suggestions or not.

**Comment by james-d-mitchell** [comment]
Thanks @wilfwilson, happy to merge your suggestions!

**Pull request merged**